### PR TITLE
[RHICOMPL-751] Autoreconnect to postgress on PG::UnableToSend

### PR DESCRIPTION
**Pull request by vkrizan**
Raw Draft

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -27,6 +27,18 @@ class Profile < ApplicationRecord
   after_rollback :destroy_orphaned_business_objective
 
   class << self
+    def find_by_sql(*args, **kwargs)
+      super(*args, **kwargs)
+    rescue ActiveRecord::StatementInvalid => e
+      if e.message =~ /^PG::UnableToSend:/
```
Comment (vkrizan) [inline app/models/profile.rb:33]: This should use `e.cause` instead.

**Comment by akofink** [comment]
This one's pretty stale. Shall we close it?

**Comment by vkrizan** [comment]
@akofink I'd like to finish this at some point. Sorry wanted to do that last Friday.

**Comment by vkrizan** [comment]
I've tried some approaches. It is possible to inject connection resets into the ORM, however it is not ideal. We should rather invest time to create a health endpoint that would report if the connection drops to let OpenShift to restart the container.

**Pull request closed without merging**